xbmc/video/VideoInfoTag.cpp

```diff
@@ -674,13 +713,22 @@ void CVideoInfoTag::ParseNative(const TiXmlElement* movie, bool prioritise)
     node = node->NextSiblingElement("actor");
   }
 
-  XMLUtils::GetString(movie, "set", m_strSet);
+  XMLUtils::GetString(movie, "set", value);
+  SetSet(value);
+
+  std::vector<std::string> tags(m_tags);
   XMLUtils::GetStringArray(movie, "tag", m_tags, prioritise, g_advancedSettings.m_videoItemSeparator);
+  SetTags(tags);
+
+  std::vector<std::string> studio(m_studio);
   XMLUtils::GetStringArray(movie, "studio", m_studio, prioritise, g_advancedSettings.m_videoItemSeparator);
+  SetStudio(studio);
```

Comment:
BTW use the handy trim support in XMLUtils + use the members directly will safe most of the lines.
The setter could be used outside, for now we can fix the JSON-RPC stuff by using the setter.